staging/src/k8s.io/kms/plugins/base64/plugin.go

```diff
@@ -0,0 +1,119 @@
+package main
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+	"os/signal"
+	"strings"
+	"sync"
+	"syscall"
+	"time"
+
+	"k8s.io/klog/v2"
+	"k8s.io/kms/pkg/hierarchy"
+	"k8s.io/kms/pkg/service"
+)
+
+var (
+	listenAddr = flag.String("listen-addr", "unix:///tmp/kms.socket", "gRPC listen address")
+	timeout    = flag.Duration("timeout", 5*time.Second, "gRPC timeout")
+)
+
+type remoteService struct {
+	mu    sync.Mutex
+	keyID string
+}
+
+func main() {
+	flag.Parse()
+
+	addr, err := parseEndpoint(*listenAddr)
+	if err != nil {
+		klog.ErrorS(err, "failed to parse endpoint")
+		os.Exit(1)
+	}
+
+	ctx := withShutdownSignal(context.Background())
+	localKEKService := hierarchy.NewLocalKEKService(
+		ctx,
+		&remoteService{
+			keyID: "base64-key-id",
+		},
```

Comment:
FYI @ritazh is currently working on a mock remote KMS. If that PR gets merged before, we might be able to switch to that and not do the base64 remote service here.

Reply:
https://github.com/kubernetes/kubernetes/pull/116031 is now merged! You can create a new remote service with `NewMockAESService` to start off and we can add the latency + rate limit layer when the CI is green.